testez-cli/src/test_place.rs

```diff
@@ -0,0 +1,122 @@
+use maplit::hashmap;
+use rbx_dom_weak::{RbxInstanceProperties, RbxTree, RbxValue};
+
+use crate::bundled_libraries;
+
+static ENTRYPOINT_PATH: &str = "ServerScriptService.TestRunner";
+
+pub struct TestPlaceInput {
+    pub source: RbxTree,
+    pub dependencies: Option<RbxTree>,
+    pub core_script_security: bool,
+}
+
+pub struct TestPlace {
+    pub tree: RbxTree,
+    pub entrypoint_path: &'static str,
+}
+
+impl TestPlace {
+    pub fn generate(mut input: TestPlaceInput) -> Self {
+        let mut tree = RbxTree::new(RbxInstanceProperties {
+            name: "Test Place".to_owned(),
+            class_name: "DataModel".to_owned(),
+            properties: Default::default(),
+        });
+
+        let game = tree.get_root_id();
+
+        let replicated_storage = tree.insert_instance(
+            RbxInstanceProperties {
+                name: "ReplicatedStorage".to_owned(),
+                class_name: "ReplicatedStorage".to_owned(),
+                properties: Default::default(),
+            },
+            game,
+        );
+
+        let source_root_id = input.source.get_root_id();
+        let source_root = input.source.get_instance(source_root_id).unwrap();
+        let source_children = source_root.get_children_ids().to_vec();
+
+        for id in source_children {
+            let instance = input.source.get_instance_mut(id).unwrap();
+            instance.properties.insert(
+                "Tags".to_owned(),
+                RbxValue::BinaryString {
+                    value: b"TestEZTestRoot".to_vec(),
+                },
+            );
```

Comment:
Is this tagging each child of the `src` folder?

Reply:
This DOM code is pretty awkward. When models are deserialized from rbx_xml, a `DataModel` instance is constructed to hold all of the top-level instances from a model. This is a concession to the fact that models can have multiple top-level instances, but I didn't want the function to return multiple disjoint `RbxTree` instances.

This code ends up copying only one directory most of the time, usually `src`.